apps/system/utils/kdbg_date.c

```diff
@@ -235,9 +299,10 @@ int kdbg_date(int argc, char **args)
 
 	if (argc == 1) {
 		ret = date_showtime();
-	} else if (!strcmp(args[1], "-s") && argc == 6) {
+	} else if (!strcmp(args[1], "-s") && (argc >= 6 && argc <= 9)) {
```

Comment:
When CONFIG_LIBC_LOCALTIME is enabled and argc is given 6, date_settime is called. But becuase of line 236, it goes failure always.